src/services/import/zip.js

```diff
@@ -240,12 +240,10 @@ async function importZip(taskContext, fileBuffer, importRootNote) {
             return /^(?:[a-z]+:)?\/\//i.test(url);
         }
 
+        content = removeTrilumTags(content);
+
         content = content.replace(/<h1>([^<]*)<\/h1>/gi, (match, text) => {
-            if (noteTitle.trim() === text.trim()) {
```

Comment:
I would still leave this as a fallback for cases where the HTML files are not coming from Trilium. In such cases this heuristic can filter useless H1s, even if it's not perfect ...